keep/api/bl/ai_suggestion_bl.py

```diff
@@ -0,0 +1,499 @@
+
+            return IncidentsClusteringSuggestion(
+                incident_suggestion=processed_incidents,
+                suggestion_id=str(suggestion.id),
+            )
+
+        except Exception as e:
+            self.logger.error(f"AI incident creation failed: {e}")
+            raise HTTPException(status_code=500, detail="AI service is unavailable.")
+
+    async def commit_incidents(
+        self,
+        suggestion_id: UUID,
+        incidents_with_feedback: List[Dict],
+        user_id: str,
+        incident_bl: IncidentBl,
+    ) -> List[IncidentDto]:
+        """Commit incidents with user feedback."""
+        committed_incidents = []
+
+        # Add feedback to the database
+        changes = {
+            incident_commit["incident"]["id"]: incident_commit["changes"]
+            for incident_commit in incidents_with_feedback
+        }
+        self.add_feedback(
+            suggestion_id=suggestion_id,
+            user_id=user_id,
+            feedback_content=changes,
+        )
+
+        for incident_with_feedback in incidents_with_feedback:
+            if not incident_with_feedback["accepted"]:
+                self.logger.info(
+                    f"Incident {incident_with_feedback['incident']['name']} rejected by user, skipping creation"
+                )
+                continue
+
+            try:
+                # Create the incident
+                incident_dto = IncidentDto.parse_obj(incident_with_feedback["incident"])
+                created_incident = incident_bl.create_incident(incident_dto)
+
+                # Add alerts to the created incident
+                alert_ids = [
+                    uuid.UUID(alert["event_id"])
+                    for alert in incident_with_feedback["incident"]["alerts"]
+                ]
+                await incident_bl.add_alerts_to_incident(created_incident.id, alert_ids)
+
+                committed_incidents.append(created_incident)
+                self.logger.info(
+                    f"Incident {incident_with_feedback['incident']['name']} created successfully"
+                )
+
+            except Exception as e:
+                self.logger.error(
+                    f"Failed to create incident {incident_with_feedback['incident']['name']}: {str(e)}"
+                )
+
+        return committed_incidents
+
+    def _prepare_prompts(
+        self, alerts_dto: List[AlertDto], topology_data: List[TopologyServiceDtoOut]
+    ) -> Tuple[str, str]:
+        """Prepare system and user prompts for AI."""
+        alert_descriptions = "\n".join(
+            [
+                f"Alert {idx+1}: {json.dumps(alert.dict())}"
+                for idx, alert in enumerate(alerts_dto)
+            ]
+        )
+
+        topology_text = "\n".join(
+            [
+                f"Topology {idx+1}: {json.dumps(topology.dict(), default=str)}"
+                for idx, topology in enumerate(topology_data)
+            ]
+        )
+
+        system_prompt = """
```

Comment:
Possible future improvement: move prompt templates to the database for easier editing